src/plugins/dashboard/public/application/top_nav/editor_menu.tsx

```diff
@@ -0,0 +1,210 @@
+          return -1;
+        }
+        if (a > b) {
+          return 1;
+        }
+        return 0;
+      })
+      .filter(({ hidden }: BaseVisType) => !hidden);
+
+  const promotedVisTypes = getVisTypesByGroup(VisGroups.PROMOTED);
+  const visTypeAliases = visualizations
+    .getAliases()
+    .sort(({ promotion: a = false }: VisTypeAlias, { promotion: b = false }: VisTypeAlias) =>
+      a === b ? 0 : a ? -1 : 1
+    );
+
+  const factories = embeddable
+    ? Array.from(embeddable.getEmbeddableFactories()).filter(
+        ({ type, isEditable, canCreateNew, isContainerType }) =>
+          isEditable() && !isContainerType && canCreateNew() && type !== 'visualization'
+      )
+    : [];
+
+  const factoryGroupMap: Record<string, FactoryGroup> = {};
+  const ungroupedFactories: EmbeddableFactoryDefinition[] = [];
+  let panelCount = 1;
+
+  factories.forEach((factory: EmbeddableFactoryDefinition, index) => {
+    const { grouping } = factory;
+
+    if (grouping) {
+      grouping.forEach((group) => {
+        if (factoryGroupMap[group.id]) {
+          factoryGroupMap[group.id].factories.push(factory);
+        } else {
+          factoryGroupMap[group.id] = {
+            id: group.id,
+            appName: group.getDisplayName ? group.getDisplayName({ embeddable }) : group.id,
+            icon: (group.getIconType ? group.getIconType({ embeddable }) : 'empty') as IconType,
+            factories: [factory],
+            panelId: panelCount,
+          };
+
+          panelCount++;
+        }
+      });
+    } else {
+      ungroupedFactories.push(factory);
+    }
+  });
+
+  const getVisTypeMenuItem = (visType: BaseVisType): EuiContextMenuPanelItemDescriptor => {
+    const { name, title, titleInWizard, description, icon = 'empty' } = visType;
+
+    return {
+      name: titleInWizard || title,
+      icon: icon as string,
+      onClick: createNewVisType(visType),
+      'data-test-subj': `visType-${name}`,
+      toolTipContent: description,
+    };
+  };
+
+  const getVisTypeAliasMenuItem = (
+    visTypeAlias: VisTypeAlias
+  ): EuiContextMenuPanelItemDescriptor => {
+    const { name, title, description, icon = 'empty' } = visTypeAlias;
+
+    return {
+      name: title,
+      icon,
+      onClick: createNewVisType(visTypeAlias),
+      'data-test-subj': `visType-${name}`,
+      toolTipContent: description,
+    };
+  };
+
+  const getEmbeddableFactoryMenuItem = (
+    factory: EmbeddableFactoryDefinition
+  ): EuiContextMenuPanelItemDescriptor => ({
+    name: factory.getDisplayName(),
+    icon: 'empty',
+    onClick: async () => {
+      if (trackUiMetric) {
+        trackUiMetric(METRIC_TYPE.CLICK, factory.type);
+      }
+      if (factory.getExplicitInput) {
+        const explicitInput = await factory.getExplicitInput();
+        await dashboardContainer.addNewEmbeddable(factory.type, explicitInput);
+      } else {
+        await factory.create({} as EmbeddableInput, dashboardContainer);
+      }
+    },
```

Comment:
Can you add a toast message here when the new embeddable is inserted?

Reply:
You got it 👍